bigsi/cmds/large_build.py

```diff
@@ -0,0 +1,40 @@
+from bigsi.storage import get_storage
+from bigsi.graph.metadata import SampleMetadata
+from bigsi.bloom import BitMatrixGroupReader
+
+BLOOM_FILTERS_SIZE_KEY = "ksi:bloomfilter_size"
+NUM_HASH_FUNCTIONS_KEY = "ksi:num_hashes"
+NUM_ROWS_KEY = "number_of_rows"
+NUM_COLS_KEY = "number_of_cols"
+DB_INSERT_BATCH_SIZE = 1000
+
+
+def large_build(config, input_path_list, num_cols_list, sample_list):
+    storage = get_storage(config)
+    num_rows = int(config["m"])
+
+    with BitMatrixGroupReader(zip(input_path_list, num_cols_list), num_rows) as bmgr:
+        processed = 0
+        bit_arrays = []
+        keys = []
+        for row_index in range(num_rows):
+            keys.append(row_index)
+            bit_arrays.append(next(bmgr))
+            processed = processed + 1
+            if processed == DB_INSERT_BATCH_SIZE:
```

Comment:
The code inside this `if` is not covered by the tests, I wonder if we could cover it (maybe move `DB_INSERT_BATCH_SIZE` as a parameter of large_build, having `1000` as default value would make it easier to test?)

Reply:
Thank you very much for pointing this out. I have refactored the test in a8f0e6c so that this path is covered.